### PR TITLE
Demonstrate Spark/Iceberg and Presto/Iceberg compatibility

### PR DESCRIPTION
**Pull request by djsstarburst**
This commit adds a number of tests to ensure that Iceberg data
written by Spark/SQL can be read by the Presto Iceberg connector,
and vice versa.  Everything worked correctly except DECIMAL, which
isn't yet supported in Spark/SQL, and TIMESTAMP, because
Iceberg doesn't yet support TIMESTAMP WITH TIME ZONE and Spark
only supports TIMESTAMP WITH TIME ZONE.

### CODE REVIEW

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java [hunk 1]
```diff
@@ -30,30 +29,177 @@
...
+
+        Row row = row(
+                "a_string",
+                1000000000000000L,
+                1000000000,
+                10000000.123F,
+                100000000000.123,
+                true,
+                "2020-06-28 14:16:00.456",
+                "1950-06-28");
```
Comment (aalbu) [inline presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java:82]: We could use `java.sql` types:
```suggestion
                Timestamp.valueOf("2020-06-28 14:16:00.456"),
                Date.valueOf("1950-06-28"));
```

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java [hunk 2]
```diff
@@ -30,30 +29,177 @@
 public class TestIcebergBasic
```
[the remark this answers is not in the document]
Comment (djsstarburst) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java:29]: Good point.  I'll change it back to that name.

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java [hunk 3]
```diff
@@ -30,30 +29,177 @@
...
+                100000000000.123,
+                true,
+                "2020-06-28 14:16:00.456",
+                "1950-06-28");
+
+        String startOfSelect = "SELECT _string, _bigint, _integer, _real, _double, _boolean";
+        QueryResult sparkSelect = onSpark().executeQuery(format("%s, CAST(_timestamp AS STRING), CAST(_date AS STRING) FROM %s", startOfSelect, sparkTableName));
+        assertThat(sparkSelect).containsOnly(row);
+
+        QueryResult prestoSelect = onPresto().executeQuery(format("%s, REPLACE(CAST(_timestamp AS VARCHAR), ' UTC'), CAST(_date AS VARCHAR) FROM %s", startOfSelect, prestoTableName(baseTableName)));
```
[the remark this answers is not in the document]
Comment (djsstarburst) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java:88]: Yes, @electrum and I spent a few minutes this morning trying to understand the best course.  @electrum is going to think about it and get back to me.

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java [hunk 4]
```diff
@@ -30,30 +29,177 @@
...
+                1000000000000000L,
+                1000000000,
+                10000000.123F,
+                100000000000.123,
+                true,
+                "2020-06-28 14:16:00.456",
+                "1950-06-28");
+
+        String startOfSelect = "SELECT _string, _bigint, _integer, _real, _double, _boolean";
+        QueryResult sparkSelect = onSpark().executeQuery(format("%s, CAST(_timestamp AS STRING), CAST(_date AS STRING) FROM %s", startOfSelect, sparkTableName));
```
[the remark this answers is not in the document]
Comment (djsstarburst) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java:85]: That's what I tried the first time but it didn't work.  Sadly, the value types returned by Spark are Hive value class instances, and the ones in the Presto are Presto value class instances.  The agree for primitive types like Long and Double, but not for `TIMESTAMP`, `TIME` and `DATE`.

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java (hunk 1 above)
Comment (djsstarburst) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java:82]: I wish it were true :)

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (djsstarburst) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergBasic.java]: @electrum, I just rebased on tip master, and re-ran the tests, and the fact Iceberg doesn't yet support `TIMESTAMP WITH TIME ZONE `and Spark only supports `TIMESTAMP WITH TIME ZONE` is still a problem, causing `testSparkReadingPrestoData()` to fail if the timestamp-related fields are uncommented in that test.

**Pull request merged**